BossMod/Modules/Shadowbringers/HuntA/Sugaar.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+
+namespace BossMod.Shadowbringers.HuntA.Sugaar
+{
+    public enum OID : uint
+    {
+        Boss = 0x2875, 
+    };
+
+    public enum AID : uint
+    {
+        AutoAttack = 870, // Boss->player, no cast, single-target
+        NumbingNoise = 18015, // Boss->self (front), 5.0s cast, range 13, 120 degree cone
+        TailSnap = 18016, // Boss->self (behind), 5.0s cast, range 18, 120 degree cone
+        BodySlam = 18018, // Boss->self, 5.0s cast, range 11 circle
+        NumbingNoiseRotation = 18100, // rotation, pulls player in from 30 by max 25 units between hitboxes, 3 attacks of NumbingNoise2
+        NumbingNoiseDuringRotation = 18098, // during rotation, 0.3s cast time, 13 range, 120 degree cone
+        TailSnapRotation = 18101, // rotation, pulls player in from 30 by max 25 units between hitboxes, 3 attacks of TailSnap2
+        TailSnapDuringRotation = 18099, // Boss->self (behind), 0.3s cast, range 18, 120 degree cone
+    }
+
+    class NumbingNoise : Components.SelfTargetedAOEs
+    {
+        public NumbingNoise() : base(ActionID.MakeSpell(AID.NumbingNoise), new AOEShapeCone(13,60.Degrees())) { } 
+    }
+    class TailSnap : Components.SelfTargetedAOEs
+    {
+        public TailSnap() : base(ActionID.MakeSpell(AID.TailSnap), new AOEShapeCone(18,60.Degrees())) { }
+    }
+    class Rotation : Components.GenericAOEs
+    {
+        private int _remainingHits = 0;
+        private int _numbingnoise = 0;
+        private int _tailsnap = 0;
+        private Angle _RotationDir;
+        private Angle _RotationDirIncrement;
+
+        private static AOEShapeCone _shapeNumbingNoise = new(13, 60.Degrees());
+        private static AOEShapeCone _shapeTailSnap = new(18, 60.Degrees());
+
+        public override IEnumerable<AOEInstance> ActiveAOEs(BossModule module, int slot, Actor actor)
+        {
+            if (_remainingHits > 0 && _numbingnoise > 0)
+                yield return new(_shapeNumbingNoise, module.PrimaryActor.Position, _RotationDir);
+            if (_remainingHits > 0 && _tailsnap > 0)
+                yield return new(_shapeTailSnap, module.PrimaryActor.Position, _RotationDir);
+        }
+
+        public override void Update(BossModule module)
+        {
+            if (module.PrimaryActor.CastInfo == null || !module.PrimaryActor.CastInfo.IsSpell())
+                return;
+            switch ((AID)module.PrimaryActor.CastInfo.Action.ID)
+            {
+                case AID.NumbingNoiseRotation:
+                    _RotationDir = module.PrimaryActor.Rotation;
+                    break;
+                case AID.TailSnapRotation:
+                    _RotationDir = module.PrimaryActor.Rotation+180.Degrees();
```

Comment:
Why do you do all this in Update, rather than in OnCastStarted?

Reply:
because you did this in the Gurangatch module and I look at existing solutions

Reply:
but I changed it